### DISCUSSION TITLE
Version 0.8.0 RC1

### DISCUSSION THREAD

**Discussion by s-allius** [s-allius/tsun-gen3-proxy#83] [Announcements]
Today I released the first release candidate for Version 0.8.0.

This version brings MODBUS support for all supported inverters. 
- Registers can be queried directly via MQTT now. 
- The proxy itself uses the MODBUS protocol to read data at a higher sampling rate and send it to the MQTT broker. This increases the time resolution but also significantly reduces the average delay.
- The `Rated Power` of the inverter can now also be set via MQTT

For GEN3PLUS inverters, this version also supports AT+ commands via MQTT. This can be used to query and set access data and WLAN parameters. 

Access lists can be used to block AT+ commands from the TSUN Cloud or separately via MQTT. By default, only firmware updates for TSUN are now enabled. 

To reduce the size of the log files for packet tracing, the proxy now uses different logging levels depending on the packet type.  Heartbeat and regular MODBUS packets are now only logged with the DEBUG log level and are therefore no longer saved in the log file by default.
 
## What's Changed
* Data collect interval is now uniformly reported  for the GEN3 and GEN3PLUS to home assistant by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/63
* The creation of the internal configuration from the config.toml file was made rebuster by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/64
* Faster `Data-Up interval` sends measurement data to the MQTT broker every minute by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/67
* Add Fifo for MODBUS requests for robustness by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/74
* Bugfix: Catch all OSError errors in the read loop by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/79
* GEN3PLUS: control access via AT commands by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/81
* trace heartbeat and regular modbus pakets with log level DEBUG by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/82

**Full Changelog**: https://github.com/s-allius/tsun-gen3-proxy/compare/v0.7.0...v0.8.0-rc.1

<hr /><em>This discussion was created from the release <a href='https://github.com/s-allius/tsun-gen3-proxy/releases/tag/v0.8.0-rc.1'>Version 0.8.0 RC1</a>.</em>

**Comment by Cardes** [comment]
Thanks a lot, i tested this release and it works just fine.
Regarding the rated_power table for a MS2000(600) with a s/n starting with Y47E, Firmware Version V4.0.20 the table looks like this:
| Rated_Load  | Rated_Power |
| -------- | ------- |
| 0  | 1600    |
| 1  | 1500    |
| 2  | 1200     |
| 3  | 600    |
| 4  | 2000 |
| 5  | 1800   |

**Reply by s-allius** [reply]
That's interesting, because the 800W are missing. What happens at Level 6? On my devices it seems, that these values will be ignored and the inverter runs unlimited with 2000W. I have on with a Y17E serial number.

 I will add your values to the wiki

**Comment by Cardes** [comment]
Oh nice, i was too caucious to try values outside the table in the wiki, at Level 6 it now says 800W.
Thanks a lot!

**Reply by s-allius** [reply]
Cool
